api/database/migrations/2024_05_21_141338_add_community_id_column_search_request_applicant_filter_tables.php

```diff
@@ -0,0 +1,90 @@
+<?php
+
+use App\Enums\PoolStream;
+use Illuminate\Database\Migrations\Migration;
+use Illuminate\Database\Schema\Blueprint;
+use Illuminate\Support\Facades\DB;
+use Illuminate\Support\Facades\Schema;
+
+return new class extends Migration
+{
+    /**
+     * Run the migrations.
+     */
+    public function up(): void
+    {
+        // Start nullable until we populate existing requests/filters
+        Schema::table('pool_candidate_search_requests', function (Blueprint $table) {
+            $table->uuid('community_id')->nullable();
+            $table->foreign('community_id')->references('id')->on('communities');
+        });
+
+        Schema::table('applicant_filters', function (Blueprint $table) {
+            $table->uuid('community_id')->nullable();
+            $table->foreign('community_id')->references('id')->on('communities');
+        });
+
+        $rows = DB::table('pool_candidate_search_requests')
+            ->join('applicant_filters', 'pool_candidate_search_requests.applicant_filter_id', 'applicant_filters.id')
+            ->select('pool_candidate_search_requests.id', 'applicant_filter_id', 'applicant_filters.qualified_streams')
+            ->get();
+
+        $ATIP = DB::table('communities')
+            ->select('id', 'key')
+            ->where('key', 'atip')
+            ->first();
+
+        $digital = DB::table('communities')
+            ->select('id', 'key')
+            ->where('key', 'digital')
+            ->first();
+
+        $rows->each(function ($row) use ($ATIP, $digital) {
```

Comment:
I think this could all be done without a foreach loop, which will be very slow given many requests. You just need to  combine your UPDATE sql statements with the join statement from line 28, and you can add a `WHERE qualified_streams` clause.

Reply:
Ahh, that is a good idea! I think I got it in [update migration to avoid looping](https://github.com/GCTC-NTGC/gc-digital-talent/pull/10432/commits/72ef8e23484231228958b516d624e70e1bd4e8af)